tests/functional/test_utilities_batch.py

```diff
@@ -115,6 +119,17 @@ def handler(record):
     return handler
 
 
+@pytest.fixture(scope="module")
+def async_record_handler() -> Callable:
```

Comment:
```suggestion
def async_record_handler() -> Callable[..., Awaitable[Any]]:
```

to make it easier to spot the return is a coroutine, not a sync return.